sunxi: add testing 5.10 kernel

abajk opened the pull request:
Run tested on: Orange PI One (cortexa7) and Orange Pi PC2 (cortexa53).
Build tested on: cortexa8

Signed-off-by: Aleksander Jan Bajkowski <A.Bajkowski@stud.elka.pw.edu.pl>

abajk committed:
sunxi: add 5.10 as testing kernel

Add 5.10 as testing kernel.

Signed-off-by: Aleksander Jan Bajkowski <A.Bajkowski@stud.elka.pw.edu.pl>

abajk committed:
sunxi: refresh patches

These patches are already in mainline 5.10 kernel:
* 410-v5.6-arm64-dts-allwinner-a64-olinuxino-Add-bank-supply-re.patch
* 420-v5.7-arm64-dts-allwinner-a64-olinuxino-add-user-red-LED.patch
* 440-add-h6-pwm.patch
* 441-arm64-dts-add-PWM-node.patch
* 443-board-h6-orangepioneplus-fix-missing-ethernet.patch

Other patches have been refreshed by executing this command:
$ make target/linux/refresh V=s

Signed-off-by: Aleksander Jan Bajkowski <A.Bajkowski@stud.elka.pw.edu.pl>

abajk committed:
sunxi: refresh 5.4 config

This was done by executing these commands:
$ make kernel_oldconfig CONFIG_TARGET=subtarget
$ make kernel_oldconfig CONFIG_TARGET=subtarget_platform

Signed-off-by: Aleksander Jan Bajkowski <A.Bajkowski@stud.elka.pw.edu.pl>

abajk committed:
kernel: move common symbol to generic config

Move some disabled config options found in sunxi target to generic.

Signed-off-by: Aleksander Jan Bajkowski <A.Bajkowski@stud.elka.pw.edu.pl>

abajk committed:
sunxi: copy files to kernel 5.10

Copy config and patches to kernel 5.10.

Signed-off-by: Aleksander Jan Bajkowski <A.Bajkowski@stud.elka.pw.edu.pl>

abajk committed:
sunxi: refresh 5.10 config

Some new symbols have been added to the target configs
and others have been moved to the subtargets.

Signed-off-by: Aleksander Jan Bajkowski <A.Bajkowski@stud.elka.pw.edu.pl>

wigyori commented:
I'll take a look at this on additional a7 and a8 boards. Did you run an all-modules-enabled build?

abajk commented:
> I'll take a look at this on additional a7 and a8 boards. Did you run an all-modules-enabled build?

Hi,
I didn't build with all modules enabled. Now I can see some modules fail to build with error:
~~~
	/home/aleksander/workspace/openwrt/bin/targets/sunxi/cortexa7/packages/kernel_5.10.18-1-2379b9eb1815a79f23bac8d6e582be04_arm_cortex-a7_neon-vfpv4.ipk
Unknown package 'kernel'.
Collected errors:
 * pkg_hash_fetch_best_installation_candidate: Packages for kernel found, but incompatible with the architectures configured
 * opkg_install_cmd: Cannot install package kernel.
~~~

adschm commented:
> Edit: Builds correctly with all kernel modules enabled. Build fails only when "Multiple devices" option is selected. I observe the same behavior on ath79 with 5.10. Building a single image works fine.

Multiple devices is what the buildbots do. It should be selected from buildbot settings checked.

abajk commented:
Changes:
* rebased to current master
* include latest upstream changes (78c1ed64fa790032dd9f8364fc4d5166ce632609 bb9915407bdf24fd45daa49b32715fe3e8e00be0)

wigyori commented:
> Have you already tested this PR on your boards?

@abajk - yes, neither of my A10/A13 boards boot up with 5.10 - trying to work out the cause.

juanesf commented:
I have compiled an image for bpi-m2ultra r40 and pinecube s3

wigyori commented:
> > Have you already tested this PR on your boards?
> 
> @abajk - yes, neither of my A10/A13 boards boot up with 5.10 - trying to work out the cause.

This is due to LPAE enabled on the cortexa8 targets - which the A8 cores don't support. I'll fix this in trunk on 5.4 (and it'll need to be backported to 21.02-rc), and will send you a small patch for 5.10 too.

wigyori commented:
> > > Have you already tested this PR on your boards?
> > 
> > 
> > @abajk - yes, neither of my A10/A13 boards boot up with 5.10 - trying to work out the cause.
> 
> This is due to LPAE enabled on the cortexa8 targets - which the A8 cores don't support. I'll fix this in trunk on 5.4 (and it'll need to be backported to 21.02-rc), and will send you a small patch for 5.10 too.

@abajk , the cortexa8/config-5.4 has been updated in trunk, if you just update the cortexa8/config-5.10 in your PR according to that, that's enough for me to give an ack this whole PR.

abajk commented:
> > > > Have you already tested this PR on your boards?
> > > 
> > > 
> > > @abajk - yes, neither of my A10/A13 boards boot up with 5.10 - trying to work out the cause.
> > 
> > 
> > This is due to LPAE enabled on the cortexa8 targets - which the A8 cores don't support. I'll fix this in trunk on 5.4 (and it'll need to be backported to 21.02-rc), and will send you a small patch for 5.10 too.
> 
> @abajk , the cortexa8/config-5.4 has been updated in trunk, if you just update the cortexa8/config-5.10 in your PR according to that, that's enough for me to give an ack this whole PR.

@wigyori
I have updated this PR with your changes.

guidosarducci commented on target/linux/generic/config-5.10:
```diff
@@ -939,6 +939,7 @@ CONFIG_CMDLINE=""
 CONFIG_COMPACTION=y
 # CONFIG_COMPAL_LAPTOP is not set
 # CONFIG_COMPAT is not set
+# CONFIG_COMPAT_32BIT_TIME is not set
```
Don't think this should be here. CONFIG_COMPAT_32BIT_TIME is currently set per target, with the default depending on the arch being 32- or 64-bit. In the future, once all userspace and libc support 64-bit time_t, this might make sense though.

abajk replied:
Thanks for the suggestion. CONFIG_COMPAT_32BIT_TIME has been removed from generic config.

abajk commented:
Recent changes:
* rebased on top of the current master
* included latest patches (5ca4330aa846bc200912e1d5c2348437a8a7514f84f2a672a3d272dc7373504a)
* dropped CONFIG_COMPAT_32BIT_TIME from generic config. Sugested by @guidosarducci

ChalesYu commented:
I suggest to add AC200 ethernet driver, It can support more ethernet port.
patches from libreelec can be found here:
[0038-mfd-Add-support-for-AC200.patch](https://github.com/LibreELEC/LibreELEC.tv/blob/master/projects/Allwinner/patches/linux/0038-mfd-Add-support-for-AC200.patch)
[0039-net-phy-Add-support-for-AC200-EPHY.patch](https://github.com/LibreELEC/LibreELEC.tv/blob/master/projects/Allwinner/patches/linux/0039-net-phy-Add-support-for-AC200-EPHY.patch)

then add `CONFIG_MFD_AC200=y` and `CONFIG_AC200_PHY=y`  to `config-5.10`.

mans0n commented on target/linux/sunxi/config-5.4:
```diff
@@ -231,59 +202,7 @@ CONFIG_HARDIRQS_SW_RESEND=y
-CONFIG_HAVE_ARCH_THREAD_STRUCT_WHITELIST=y
-CONFIG_HAVE_ARCH_TRACEHOOK=y
-CONFIG_HAVE_ARCH_TRANSPARENT_HUGEPAGE=y
-CONFIG_HAVE_ARM_ARCH_TIMER=y
```
I don't know what this exactly does, but it seems that removing this option causes a change in `build_dir/target-*/linux-sunxi_*/linux-5.10.*/.config`. I'm not sure this is an intended behavior or not...

mans0n commented on target/linux/sunxi/config-5.10:
```diff
@@ -430,6 +434,7 @@ CONFIG_SUN8I_DE2_CCU=y
 CONFIG_SUN8I_H3_CCU=y
 CONFIG_SUN8I_R40_CCU=y
 CONFIG_SUN8I_R_CCU=y
+# CONFIG_SUN8I_THERMAL is not set
```
I think we can enable this at least for a7 and a53, what do you think?
In case of crypto drivers, we may enable them by adding kernel modules later.

mans0n commented on target/linux/sunxi/config-5.10:
```diff
@@ -376,7 +380,6 @@ CONFIG_RESET_SUNXI=y
 CONFIG_RFS_ACCEL=y
 CONFIG_RPS=y
 CONFIG_RWSEM_SPIN_ON_OWNER=y
-CONFIG_SATA_PMP=y
```
Note that this is also a functional regression. #4292
But I don't think we need to take care of it right now...(until someone yells at it)

mans0n reviewed (approved):
Looks good to me.

mans0n commented:
With that,
`Reviewed-by: Sungbo Eo <mans0n@gorani.run>`

I haven't tested this branch, but I've also made my own 5.10 patches (which turned out to be very similar to this PR) and run-tested on PinePhone and then I ran into this... :')

abajk commented:
Openwrt got 5.10 support for sunxi in the commit 	83672f506d95fe3d79b5afef6595e946ceeb4443
